packages/open-next/src/adapters/routing/matcher.ts

```diff
@@ -22,14 +22,14 @@ const routeHasMatcher =
     switch (redirect.type) {
       case "header":
         return (
-          headers?.[redirect.key.toLowerCase()] !== "" &&
```

Comment:
@conico974 Should this be updated too?

Reply:
This probably needs to be updated as well, but i think we should do something like 
```ts
headers?.[redirect.key.toLowerCase()] && headers?.[redirect.key.toLowerCase()] !== ""
```

Reply:
`!!headers?.[redirect.key.toLowerCase()]` should take care of both those conditions right? `undefined` and empty string will both be false.

Reply:
@conico974 gonna merge this unless I should make your suggested changes.